### ISSUE TITLE
Webinterface not working, running under Fedora 31

### ISSUE THREAD

**Issue by arno1979** [pyload/pyload#3488]
### Describe the problem

Yesterday I updated my server operating system from Fedora 29 to 31.
Since then I cannot access the webinterface anymore. Neither https://domain.tld:8001 or my subdomain using a reverse proxy are working anymore.

After the update I made sure the necessary python2 packages were installed. I am currently running version 2.7.17.
I start pyload using the following command:
`/usr/bin/python2 /usr/local/bin/pyload-stable/pyLoadCore.py --configdir=/root/.pyload/`

That gives me the following output:

```
10.12.2019 11:13:12 INFO      Starting pyLoad 0.4.9
10.12.2019 11:13:12 INFO      Using home directory: /root/.pyload
10.12.2019 11:13:12 INFO      Activated plugins: Checksum, ClickNLoad, EventMapper, ExternalScripts, UnSkipOnFail, UpdateManager, UserAgentSwitcher, XFileSharing
10.12.2019 11:13:12 INFO      Deactivate plugins: AndroidPhoneNotify, AntiStandby, AntiVirus, AppriseNotify, BypassCaptcha, Captcha9Kw, CloudFlareDdos, DeathByCaptcha, DeleteFinished, DiscordNotifier, DownloadScheduler, ExpertDecoders, ExtractArchive, HotFolder, IRC, ImageTyperz, JustPremium, LinkFilter, LinkdecrypterComHook, LogMarker, MergeFiles, MultiHome, PushBullet, PushOver, RestartFailed, SkipRev, TransmissionRPC, WindowsPhoneNotify, XMPP
10.12.2019 11:13:12 INFO      Downloadtime: True
10.12.2019 11:13:12 INFO      Free space: 97.24 GiB
10.12.2019 11:13:12 INFO      Starting threaded SSL webserver: 0.0.0.0:8001
10.12.2019 11:13:12 INFO      Activating Accounts...
10.12.2019 11:13:13 INFO      Activating Plugins...
10.12.2019 11:13:13 INFO      ADDON XFileSharing: Auto-discover new hosters
10.12.2019 11:13:13 INFO      ADDON XFileSharing: Auto-discover new crypters
10.12.2019 11:13:13 INFO      pyLoad is up and running
10.12.2019 11:13:13 INFO      ADDON UpdateManager: pyLoad is up to date!
10.12.2019 11:13:13 INFO      ADDON UpdateManager: All plugins are up to date!
```

As soon as I try to access the webinterface in a browser the following messages appear:
```
Exception in thread Thread-5:
Traceback (most recent call last):
  File "/usr/lib64/python2.7/threading.py", line 804, in __bootstrap_inner
    self.run()
  File "/usr/local/bin/pyload-stable/module/web/ServerThread.py", line 71, in run
    self.start_threaded()
  File "/usr/local/bin/pyload-stable/module/web/ServerThread.py", line 93, in start_threaded
    webinterface.run_threaded(host=self.host, port=self.port, cert=self.cert, key=self.key)
  File "/usr/local/bin/pyload-stable/module/web/webinterface.py", line 156, in run_threaded
    run(app=web, host=host, port=port, server=CherryPyWSGI, quiet=True)
  File "/usr/local/bin/pyload-stable/module/lib/bottle.py", line 2383, in run
    server.run(app)
  File "/usr/local/bin/pyload-stable/module/web/utils.py", line 137, in run
    server.start()
  File "/usr/local/bin/pyload-stable/module/lib/wsgiserver/__init__.py", line 1607, in start
    self.tick()
  File "/usr/local/bin/pyload-stable/module/lib/wsgiserver/__init__.py", line 1678, in tick
    conn = self.ConnectionClass(s, self.wsgi_app, environ)
  File "/usr/local/bin/pyload-stable/module/lib/wsgiserver/__init__.py", line 1142, in __init__
    if SSL and isinstance(sock, SSL.ConnectionType):
AttributeError: 'module' object has no attribute 'ConnectionType'
```
I am not very experienced using python but of course I suspect that something has changed with the update to the lastest Fedora version. Especially since python2 support has been dropped from this release. Maybe downgrading to another python2 version could help?

If it is of any help my virtual host configuration for the reverse proxy running under a subdomain looks like this:
```
###Pyload###
<VirtualHost *:80>
    ServerName pyload.domain.tld
    ServerAlias www.pyload.domain.tld
    RedirectMatch permanent ^/(.*) https://pyload.domain.tld/$1
</VirtualHost>

<VirtualHost *:443>
    ServerName pyload.domain.tld
    ServerAlias www.pyload.domain.tld
    ProxyPreserveHost On

    # setup the proxy
    <Proxy *>
        Order allow,deny
        Allow from all
    </Proxy>
    ProxyPass / https://localhost:8001/
    ProxyPassReverse / https://localhost:8001/
</VirtualHost>
```

Any help would be really appreciated. I love pyload!

### Debug log

Running in Debug mode is giving the same error output as stated above.

**Comment by GammaC0de** [comment]
Can you run this code under python2?
```
from OpenSSL import SSL
SSL.ConnectionType
```

**Comment by arno1979** [comment]
Hi, thanks for getting back to me so quickly!

On a root login shell I get this:
```
[root@kanfansel ~]# python2
Python 2.7.17 (default, Oct 20 2019, 00:00:00) 
[GCC 9.2.1 20190827 (Red Hat 9.2.1-1)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from OpenSSL import SSL
>>> SSL.ConnectionType
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'module' object has no attribute 'ConnectionType'
>>> 
```

**Comment by GammaC0de** [comment]
can you try installing pycrypro (`pip2 install pycrypto`) and then retry the above code?

**Comment by arno1979** [comment]
I already have pyOpenSSL installed so this doesn't change anything:
```
Requirement already satisfied: pyOpenSSL in /usr/lib64/python2.7/site-packages (19.1.0)
Requirement already satisfied: cryptography>=2.8 in /usr/lib64/python2.7/site-packages (from pyOpenSSL) (2.8)
Requirement already satisfied: six>=1.5.2 in /usr/lib/python2.7/site-packages (from pyOpenSSL) (1.12.0)
Requirement already satisfied: enum34; python_version < "3" in /usr/lib/python2.7/site-packages (from cryptography>=2.8->pyOpenSSL) (1.1.6)
Requirement already satisfied: cffi!=1.11.3,>=1.8 in /usr/lib64/python2.7/site-packages (from cryptography>=2.8->pyOpenSSL) (1.13.2)
Requirement already satisfied: ipaddress; python_version < "3" in /usr/lib/python2.7/site-packages (from cryptography>=2.8->pyOpenSSL) (1.0.18)
Requirement already satisfied: pycparser in /usr/lib/python2.7/site-packages (from cffi!=1.11.3,>=1.8->cryptography>=2.8->pyOpenSSL) (2.19)
```

**Comment by GammaC0de** [comment]
What happens if you edit `/usr/local/bin/pyload-stable/module/lib/wsgiserver/__init__.py` and do search and replace `ConnectionType` with `Connection`?
(don`t forget to backup first)

**Comment by arno1979** [comment]
That actually works! Thanks so much for your help.
But these changes will be lost if there ever is an update, right?
And I just found the reason why:
If we look at https://pypi.org/project/pyOpenSSL/ under the release information for the latest version 19.1.0 (released on 2019-11-18) we can see that the alias "ConnectionType" was removed.
So I guess this will have to be fixed on the pyload side.

Thanks for your help!
